ci/install-linux.sh

```diff
@@ -16,19 +16,33 @@
 
 set -eu
 
+# Create a Docker image with all the dependencies necessary to build the
+# project.
 if [ "${TRAVIS_OS_NAME}" != "linux" ]; then
-  echo "Not a Linux-based build; skipping Docker installation and restore from cache."
+  echo "Not a Linux-based build; skipping Docker image creation."
   exit 0
 fi
 
-if [ "${CI:-}" = "true" ]; then
-  # Running in the CI environment, upgrade Docker.
-  sudo apt-get update
-  sudo apt-get install -y docker-ce
-  sudo docker --version
-fi
-
 readonly IMAGE="cached-${DISTRO}-${DISTRO_VERSION}"
-sudo docker build -t "${IMAGE}:tip" \
-     --build-arg DISTRO_VERSION="${DISTRO_VERSION}" \
-     -f "ci/Dockerfile.${DISTRO}" ci
+# Make three attempts to build the Docker image.  From time to time the image
+# creation fails because some apt server times out, and more often than not
+# restarting the build is successful.
+
+# Initially wait at least 3 minutes (the times are in seconds), because it makes
+# no sense to try faster.  We currently have 20+ minutes of remaining budget to
+# complete a build, and we should be as nice as possible to the servers that
```

Comment:
Hmm, I thought on GCE, Ubuntu apt servers are mapped to GCS buckets... or at least servers running on GCE? Unfortunate that they're timing out; maybe a bug that needs to be raised to appropriate team.

Reply:
These are the public (from github.com) Docker images for Ubuntu, Fedora, and CentOS.